firmware_tools/ghidra/vxhunter_utility/symbol.py

```diff
@@ -185,58 +193,65 @@ def demangled_symbol(symbol_string):
                 sym_demangled = demangler.demangle(symbol_string, False)
 
         except DemangledException as err:
-            logger.debug("DemangledException: symbol_string: {}, reason:{}".format(symbol_string, err))
+            logger.debug("First pass demangling failed: symbol_string: {}, reason: {}".format(symbol_string, err))
+            pass
 
-        try:
-            if not sym_demangled:
+        if not sym_demangled:
+            try:
                 # Temp fix to handle _ prefix function name by remove _ prefix before demangle
                 sym_demangled = demangler.demangle(symbol_string[1:], False)
 
-        except DemangledException as err:
-            logger.debug("DemangledException: symbol_string: {}, reason:{}".format(symbol_string, err))
+            except DemangledException as err:
+                logger.debug("Second pass demangling failed: symbol_string: {}, reason:{}".format(symbol_string, err))
+                pass
 
         if sym_demangled:
             sym_demangled_name = sym_demangled.getSignature(False)
 
-        if sym_demangled_name:
-            logger.debug("sym_demangled_name: {}".format(sym_demangled_name))
+            if sym_demangled_name:
+                logger.debug("sym_demangled_name: {}".format(sym_demangled_name))
+            else:
+                logger.debug("Demangled symbol name for string {} is None.".format(symbol_string))
 
     return sym_demangled_name
 
 
 def add_symbol(symbol_name, symbol_name_address, symbol_address, symbol_type):
     symbol_address = toAddr(symbol_address)
     symbol_name_string = symbol_name
-
     # Get symbol_name
     if symbol_name_address:
```

Comment:
We need call `removeDataAt(symbol_name_address)` before `createAsciiString` otherwise lot's of CodeUnitInsertionException will raise.

```suggestion
    if symbol_name_address:
        symbol_name_address = toAddr(symbol_name_address)
        logger.debug("Have symbol name {} at address {}.".format(symbol_name_string, symbol_name_address))

        if getDataAt(symbol_name_address):
            logger.debug("Data detected at {}; removing to make room for symbol {}".format(symbol_name_address, symbol_name))
            removeDataAt(symbol_name_address)
        else:
            logger.debug("No data detected at {}. Moving on...".format(symbol_address))

        try:
            symbol_name_string = createAsciiString(symbol_name_address).getValue()
            logger.debug("Created ascii string {} at {}.".format(symbol_name_string, symbol_name_address))
        except CodeUnitInsertionException as err:
            logger.error("Failed to create ascii string for symbol named {} at {}: {}".format(symbol_name, symbol_name_address, err))
        except BaseException as err:
            logger.error("Failed to create ascii string for symbol named {} at {}: {}; returning.".format(symbol_name, symbol_name_address, err))
            return
```

Reply:
Oh my bad! I misinterpreted.

Reply:
Okay -- I've restored it to where it once was.